plugins/outputs/nats/nats.go

```diff
@@ -85,7 +96,33 @@ func (n *NATS) Connect() error {
 
 	// try and connect
 	n.conn, err = nats.Connect(strings.Join(n.Servers, ","), opts...)
+	if n.Jetstream != nil {
+		// connect to jetstream
+		n.jetstreamClient, err = jetstream.New(n.conn)
+		if err != nil {
+			return fmt.Errorf("failed to connect to jetstream: %w", err)
+		}
+		if n.Jetstream.AutoCreateStream && !n.streamExists(n.Jetstream.StreamConfig.Name) {
+			err = choice.Check(n.Subject, n.Jetstream.StreamConfig.Subjects)
+			if err != nil {
+				return errors.New("jetstream subjects array does not contain NATS subject")
```

Comment:
What is the source of this limitation?

Reply:
Many subjects map to a stream. However publish API is for just one subject. So to reuse the field already defined for nats core, we can enforce this validation. One of the subjects has to match with the subject, since telegraf uses that field in the publishMsg function.

Let me know if you have better suggestions

Reply:
> So to reuse the field already defined for nats core, we can enforce this validation.

You lost me here. Why do we want to reuse the field? I see we use the top level subject field to do the write, but then what is the point of the streamconfig's subjects and where is it ever used?

Reply:
I have modified this part now. The validation is not done anymore. Instead, telegraf will use the subject and the stream name to create the stream config.